### DISCUSSION TITLE
Can't run mtr from install

### DISCUSSION THREAD

**Discussion by urbanracer34** [Homebrew/discussions#5311] [Getting started] [answered]
### Output of `brew config`

```shell
Last login: Tue Apr 23 20:05:13 on ttys001

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
JeradsRetinaMBP:~ jerad$ brew config
HOMEBREW_VERSION: 4.2.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 54bea636b41f7953c01a7da5f488f08c1b42ffd1
Last commit: 2 days ago
Core tap JSON: 24 Apr 02:08 UTC
Core cask tap JSON: 24 Apr 02:08 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.4zQDZxlmgZ/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-x86_64
```

### Output of `brew doctor`

```shell
JeradsRetinaMBP:~ jerad$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
JeradsRetinaMBP:~ jerad$
```

### Description of issue

I can't seem to get mtr to run, with . I ran doctor and installed the new CLI tools as they were out of date. I also had the below error which I ran the remove command for:

Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core

I've tried reinstalling mtr multiple times. 

My preferred shell is bash as that is what I am used to.

Can I get some help please?

**Comment by SMillerDev** [comment]
What does `brew ls mtr` say?

**Reply by urbanracer34** [reply]
Last login: Tue Apr 23 21:02:55 on ttys000

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
JeradsRetinaMBP:~ jerad$ brew ls mtr
/usr/local/Cellar/mtr/0.95/sbin/mtr
/usr/local/Cellar/mtr/0.95/sbin/mtr-packet
/usr/local/Cellar/mtr/0.95/share/bash-completion/completions/mtr
/usr/local/Cellar/mtr/0.95/share/man/ (2 files)
JeradsRetinaMBP:~ jerad$

**Reply by SMillerDev** [reply]
So if you fix this warning from brew doctor it should work:
```console
Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

**Reply by urbanracer34** [reply]
I am on bash. I don't think that command will work.

**Reply by SMillerDev** [reply]
You can replace zshrc with bashrc and it should work. I mostly copied the warning to indicate the problem

**Reply by urbanracer34** [reply]
Here's me running the command above and trying to run mtr after

Last login: Tue Apr 23 21:02:55 on ttys000

```
The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
JeradsRetinaMBP:~ jerad$ brew ls mtr
/usr/local/Cellar/mtr/0.95/sbin/mtr
/usr/local/Cellar/mtr/0.95/sbin/mtr-packet
/usr/local/Cellar/mtr/0.95/share/bash-completion/completions/mtr
/usr/local/Cellar/mtr/0.95/share/man/ (2 files)
JeradsRetinaMBP:~ jerad$ echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.bashrc
JeradsRetinaMBP:~ jerad$ sudo mtr reddit.com
Password:
sudo: mtr: command not found
JeradsRetinaMBP:~ jerad$ 
```

**Reply by SMillerDev** [reply]
Did you source your bashrc file after? That warning is really there to give you an idea what the issue is and isn't a complete guide to resolving all issues. Since you are already using the terminal you have two options to apply these changes for MTR to work:
1. Update your bash config by doing `source ~/.bashrc`
2. Apply the same changes to the currently running terminal by doing `export PATH="/usr/local/sbin:$PATH"`

**Reply by urbanracer34** [reply]
I'm not sure how to do either of those solutions. Can you help me out? The first option I think is the best one. And what do you mean by "sourcing" the file? 

EDIT: I did the 2nd option and it worked right away! 

EDIT2: Closed out of terminal and reopened. Failed to run the app again. I'm back at square one!

**Reply by SMillerDev** [reply]
Does `~/.bashrc` have anything in it besides that line the command added?

**Reply by urbanracer34** [reply]
Contents below: 

`export PATH="/usr/local/sbin:$PATH"`
`export PATH="/usr/local/sbin:$PATH"
`

**Reply by SMillerDev** [reply]
Do you have a `~/.bash_profile`?

**Reply by urbanracer34** [reply]
Yes I do but it is just an alias to ssh to a server which no longer exists.

**Reply by SMillerDev** [reply, marked as answer]
Try putting the `export PATH="/usr/local/sbin:$PATH"` line in there.

**Reply by urbanracer34** [reply]
That worked! Thank you so much for the help!

**Reply by gromgit** [reply]
> Closed out of terminal and reopened. Failed to run the app again. I'm back at square one!

Since `brew config` says you're running macOS, that `export` line (and there should only be one line) probably needs to be in `~/.bash_profile` instead of `~/.bashrc`, because the Terminal app you're running defaults to launching `bash` as an interactive login shell.

**Reply by urbanracer34** [reply]
@gromgit  @SMillerDev gave me all the info I needed to make it work.

**Comment by urbanracer34** [comment]
My problem has been solved.

**Comment by AndreRoode** [comment]
Good day,

I’m also experiencing issues with mtr. Entering `PATH=$PATH:/opt/homebrew/sbin` in the terminal is the only way I’ve found to make mtr work. However, it stops working every time I close and reopen the terminal. This has been a recurring issue over the past few months, and I’m hoping someone can suggest a permanent solution.

**brew config:**

HOMEBREW_VERSION: 4.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 91f3671b16fe62121364c465c14e0dfd088d7ead
Last commit: 3 days ago
Core tap JSON: 30 Oct 19:34 UTC
Core cask tap JSON: 30 Oct 19:34 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.47.0 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: N/A
Rosetta 2: false

I also get the following with brew doctor:

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc

I have found that `echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc` does not do anything if I run it in my terminal.

Please let me know if you need any more information from my side—or if I’m simply overlooking something obvious.

**Reply by gromgit** [reply]
> I have found that `echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc` does not do anything if I run it in my terminal.

Did you start a new terminal session afterwards? The above adds the necessary command, but that command only gets runs in shell sessions started _after_ it's added.

**Reply by AndreRoode** [reply]
Yes, I have closed the terminal session after and I still get "zsh: command not found: mtr".

**Reply by gromgit** [reply]
Run the following:
```
zsh -xlic "echo exit >/dev/tty" 2>&1 | tee /tmp/trace.log
```
then look at the contents of `/tmp/trace.log`. That will show you the exact series of commands that are run when you start a new terminal session, along with the filenames and line numbers where they can be found.

I'm guessing that your `PATH` is being reset by something after it's been set properly for Homebrew.

**Reply by AndreRoode** [reply]
Thank you, 
Will you be able to help me look for what is causing this as I am not sure what PATH will be correct and what will be wrong.

**Reply by gromgit** [reply]
You'll have to upload the `trace.log` file to this thread.

**Reply by AndreRoode** [reply]
Thank you
[trace.log](https://github.com/user-attachments/files/17590086/trace.log)

**Reply by gromgit** [reply]
Yup, you have a problem:
```
+/Users/andreroode/.zprofile:1> /opt/homebrew/bin/brew shellenv
+/Users/andreroode/.zprofile:1> eval export 'HOMEBREW_PREFIX="/opt/homebrew";' export 'HOMEBREW_CELLAR="/opt/homebrew/Cellar";' export 'HOMEBREW_REPOSITORY="/opt/homebrew";' 'fpath[1,0]="/opt/homebrew/share/zsh/site-functions";' 'PATH="/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Applications/VMware' 'Fusion.app/Contents/Public:/Applications/VMware";' export 'PATH;' '[' -z '"${MANPATH-}"' ']' '||' export 'MANPATH=":${MANPATH#:}";' export 'INFOPATH="/opt/homebrew/share/info:${INFOPATH:-}";'
```
you run `brew shellenv` as directed in `~/.zprofile` line 1, and it Does The Right Thing, adding both `/opt/homebrew/bin` and `/opt/homebrew/sbin` (in the order which is generally expected and desired)...
```
+/Users/andreroode/.zshrc:1> export PATH='/opt/homebrew/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Applications/VMware Fusion.app/Contents/Public:/Applications/VMware'
+/Users/andreroode/.zshrc:2> export PATH=/opt/homebrew/bin:/opt/homebrew/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Applications/VMware Fusion.app/Contents/Public
/Users/andreroode/.zshrc:export:2: not valid in this context: Fusion.app/Contents/Public
```
but then you reset your PATH twice in `~/.zshrc`, the first of which unnecessarily reverses the order of the Homebrew directories, and the second removes `/opt/homebrew/sbin` and replaces it with a redundant copy of `/opt/homebrew/bin`. (The last line I quoted about also indicates a syntax error in line 2, because there's a path component with a space in it, but you forgot to quote the PATH value.)

Unless I missed something, both lines in `~/.zshrc` don't add any new PATH components, and are therefore redundant. If you agree, I suggest replacing those two lines with the `brew shellenv` lines from `~/.zprofile`.

**Reply by AndreRoode** [reply]
Thank you for your assistance. I have made the changes to `~/.zshrc` and now mtr is working.
